src/crh/sha384/mod.rs

```diff
@@ -0,0 +1,81 @@
+use crate::crh::{CRHScheme, TwoToOneCRHScheme};
+use crate::{Error, Vec};
+
+use ark_std::rand::Rng;
+
+// Re-export the RustCrypto Sha384 type and its associated traits
+pub use sha2::{digest, Sha384};
+
+#[cfg(feature = "r1cs")]
+mod r1cs_utils;
+
+#[cfg(feature = "r1cs")]
+pub mod constraints;
+
+// Implement the CRH traits for SHA-384
+
+use core::borrow::Borrow;
+use sha2::digest::Digest;
+
+impl CRHScheme for Sha384 {
+    type Input = [u8];
+    // This is always 32 bytes. It has to be a Vec to impl CanonicalSerialize
+    type Output = Vec<u8>;
```

Comment:
This should work in `0.4`

```suggestion
    type Output = [u8; 32];
```

Reply:
Seems like I forgot to change the comment here on both SHA384 and SHA512. SHA384 has an output size of 48 bytes and SHA512 an output size of 64. Fixed these now in the new commit.